File: src/main/java/jb/data/ClfObjectOnceSorted.java
package jb.data;

public class ClfObjectOnceSorted extends ClfObject {

    public ClfObjectOnceSorted(double[] x, int y) {
        super(x, y);
    }

    @Override
    public int compareTo(ClfObject o) {
        return Double.compare(x[0], o.getX()[0]);
    }

    public ClfObjectDoubleSorted convertToDoubleSorted() {
        return new ClfObjectDoubleSorted(this.getX(), this.getY());
    }

}

File: src/main/java/jb/data/clfobj/ClfObjectOnceSorted.kt
package jb.data.clfobj

class ClfObjectOnceSorted(x: DoubleArray, y: Int) : ClfObject(x, y) {

    override fun compareTo(o: ClfObject): Int {
        return java.lang.Double.compare(x[0], o.x[0])
    }

    fun convertToDoubleSorted(): ClfObjectDoubleSorted {
        return ClfObjectDoubleSorted(this.x, this.y)
    }

}
